src/Data/Generic/Rep.purs

```diff
@@ -10,25 +10,46 @@ module Data.Generic.Rep
   , Argument(..)
   ) where
 
+import Data.Semigroup ((<>))
+import Data.Show (class Show, show)
+import Data.Symbol (class IsSymbol, reflectSymbol)
+import Type.Proxy (Proxy(..))
+
 -- | A representation for types with no constructors.
 data NoConstructors
 
 -- | A representation for constructors with no arguments.
 data NoArguments = NoArguments
 
+instance showNoArguments :: Show NoArguments where
+  show _ = "NoArguments"
+
 -- | A representation for types with multiple constructors.
 data Sum a b = Inl a | Inr b
 
+instance showSum :: (Show a, Show b) => Show (Sum a b) where
+  show (Inl a) = "(Inl " <> show a <> ")"
+  show (Inr b) = "(Inr " <> show b <> ")"
+
 -- | A representation for constructors with multiple fields.
 data Product a b = Product a b
 
+instance showProduct :: (Show a, Show b) => Show (Product a b) where
+  show (Product a b) = "(Product " <> show a <> " " <> show b <> ")"
+
 -- | A representation for constructors which includes the data constructor name
 -- | as a type-level string.
 newtype Constructor (name :: Symbol) a = Constructor a
 
+instance showConstructor :: (IsSymbol name, Show a) => Show (Constructor name a) where
+  show (Constructor a) = "(Constructor " <> show (reflectSymbol (Proxy :: Proxy name)) <> " " <> show a <> ")"
```

Comment:
Can I request that we put a `@` symbol before the constructor name? It wouldn’t be valid syntax (we don’t have explicit type applications) but it does at least clarify that the string isn’t an actual constructor argument for Constructor.

Reply:
Meaning this? Yeah, I think that makes sense.
```diff
- "(Constructor " <> show (reflectSymbol ...
+ "(Constructor @" <> show (reflectSymbol ...
```

Reply:
Yeah, exactly that.

Reply:
Fixed.